### PR TITLE
Fix `pandas` `FutureWarning`

### LINKED ISSUE #141: Address pandas FutureWarning
Raised in https://github.com/brainglobe/brainreg/actions/runs/6079643446

```
FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
```

### PR DESCRIPTION
**Pull request by willGraham01**
## Description

**What is this PR**

- [x] Bug fix
- [ ] Addition of a new feature
- [ ] Other

**Why is this PR needed?**

CI is failing without fix.
Fixes this issue before creating the release of `brainreg` for `brainglobe-v1`.

**What does this PR do?**

Closes #141.

## How has this PR been tested?

CI tests pass locally, with tox.

## Checklist:

- [x] The code has been tested locally
- [ ] Tests have been added to cover all new functionality (unit & integration)
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: tests/tests/test_unit/test_misc.py
Comment (alessandrofelder) [inline tests/tests/test_unit/test_misc.py:22]:
```suggestion
```
Is this needed? I don't think I've seen it in tests (or elsewhere)?

**Review by alessandrofelder — APPROVED** [review]
:rocket:

File: tests/tests/test_unit/test_misc.py
Comment (willGraham01) [reply tests/tests/test_unit/test_misc.py:22]: Probably a case of VSCode / intellisense being a bit too clever :sweat_smile:

**Review by alessandrofelder — APPROVED** [review]
:+1:

(BTW, if a reviewer comments and approve, they trust you to consider their comments and then merge without needing another look, as (not entirely clearly :grimacing: ) documented in https://brainglobe.info/developers/index.html#pull-requests

**Comment by willGraham01** [comment]
> (BTW, if a reviewer comments and approve, they trust you to consider their comments and then merge without needing another look, as (not entirely clearly 😬 ) documented in https://brainglobe.info/developers/index.html#pull-requests

That's what I thought but I think this repo requires an up-to-date review before merging. (I hit commit suggestion and then it dismissed your review and merging was blocked :sweat_smile:) 
Should I disable this in the repo settings?

**Comment by alessandrofelder** [comment]
Ah, I see... I'll update in the repo settings

**Pull request merged**
